File: ShopPositions/solve/java/ShopPositionsSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class ShopPositionsSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int n = (Integer) reader.next(Integer.class);
        reader.next();
        
        int m = (Integer) reader.next(Integer.class);
        reader.next();
        
        List<Integer> cBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] c = new int[cBoxed.size()];
        for (int _i = 0; _i < cBoxed.size(); ++_i)
            c[_i] = cBoxed.get(_i);
        reader.close();

        ShopPositions solver = new ShopPositions();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.maxProfit(n, m, c));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}

File: ShopPositions/solve/cpp/ShopPositionsSolver.cpp
#include <fstream>
#include <topcoder>
#include "ShopPositions.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int n; tc::read(ifs, n); tc::next(ifs);
    int m; tc::read(ifs, m); tc::next(ifs);
    vector<int> c; tc::read(ifs, c);
    ifs.close();

    std::ofstream ofs(argv[2]);
    ShopPositions solver;
    tc::show(ofs, solver.maxProfit(n, m, c));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
